elytron/WFLY-13762_ssl_context_based_on_peer_info.adoc

```diff
@@ -0,0 +1,96 @@
+= WFLY-13762 SSLContext to support delegation to alternate instances based on peer information
+:author:            Diana Vilkolakova
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+As an SSLEngine is created the host and port information of the peer being accessed is available at the time the engine is requested:
+
+https://download.java.net/java/early_access/jdk11/docs/api/java.base/javax/net/ssl/SSLContext.html#createSSLEngine(java.lang.String,int)
+
+This feature is about making use of this information to dynamically select alternative configurations of SSL contexts.
+
+This feature deals with outbound requests.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-13762[WFLY-13762]
+
+=== Related Issues
+
+* follow up on: https://issues.redhat.com/browse/EAP7-1096[EAP7-1096]
+* https://issues.redhat.com/browse/EAP7-1121[EAP7-1121]
+* https://issues.redhat.com/browse/EAP7-1121[EAP7-1120]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
```

Comment:
Is this feature going to be testes by QE or Engineering, btw?

Reply:
> Is this feature going to be tested by QE or Engineering, btw?

Undecided. It will be discussed with the QE contact when assigned.